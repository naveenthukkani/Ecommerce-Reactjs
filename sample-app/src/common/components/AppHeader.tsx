import React from 'react';
import { Link } from 'react-router-dom';
import "./Style.css"

const screenNames ={
	home:"Home",
	myorders: "My Orders",
	cart: "Cart",
	bookdetails: "Book Details"
}

export const AppHeader= () => {
	let screenName = window.location.pathname.replace("/","")
	return (
	<div className="Main-header-container">
		<div className="Title">eCommerce Site {screenName !==""? "|" :""} {screenName}</div>
		<div className="Header-section">
		<Link to="/home"><span>Home</span></Link><span> | </span>
			<Link to="/myorders"><span>My Orders</span></Link><span> | </span>
			<Link to="/cart"><span>Cart</span></Link>
		</div>
	</div>
)
}