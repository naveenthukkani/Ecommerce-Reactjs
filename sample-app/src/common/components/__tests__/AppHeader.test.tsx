import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {AppHeader} from '../AppHeader';

describe("Header", () =>{

  it('renders header component correctly', () => {
    const component = shallow(<AppHeader />);
    expect(toJson(component)).toMatchSnapshot();
  });

});