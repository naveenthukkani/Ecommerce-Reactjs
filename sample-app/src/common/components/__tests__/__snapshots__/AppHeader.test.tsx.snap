// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders header component correctly 1`] = `
<div
  className="Main-header-container"
>
  <div
    className="Title"
  >
    eCommerce Site 
     
  </div>
  <div
    className="Header-section"
  >
    <Link
      to="/home"
    >
      <span>
        Home
      </span>
    </Link>
    <span>
       | 
    </span>
    <Link
      to="/myorders"
    >
      <span>
        My Orders
      </span>
    </Link>
    <span>
       | 
    </span>
    <Link
      to="/cart"
    >
      <span>
        Cart
      </span>
    </Link>
  </div>
</div>
`;
